dht.go

```diff
@@ -275,7 +276,49 @@ func (dht *IpfsDHT) putLocal(key string, rec *recpb.Record) error {
 // on the given peer.
 func (dht *IpfsDHT) Update(ctx context.Context, p peer.ID) {
 	logger.Event(ctx, "updatePeer", p)
-	dht.routingTable.Update(p)
+	if dht.shouldAddPeerToRoutingTable(p) {
+		dht.routingTable.Update(p)
+	}
+}
+
+func (dht *IpfsDHT) shouldAddPeerToRoutingTable(p peer.ID) bool {
+	return dht.hasOutboundConnToPeer(p) ||
+		dht.hasSensibleAddressesForPeer(p)
+}
+
+func (dht *IpfsDHT) hasOutboundConnToPeer(p peer.ID) bool {
+	cons := dht.host.Network().ConnsToPeer(p)
+	for _, c := range cons {
+		if c.Stat().Direction == network.DirOutbound {
+			return true
+		}
+	}
+	return false
+}
+
+func (dht *IpfsDHT) hasSensibleAddressesForPeer(p peer.ID) bool {
+	if dht.peerIsOnSameSubnet(p) {
+		// TODO: for now, we can't easily tell if the peer on our subnet
+		// is dialable or not, so don't discriminate.
```

Comment:
Hack idea: use `Discovery.MDNS.Enabled` for this

Reply:
edit: nevermind, afetr reading rest of the code, this comment doesn't make sense